### PR TITLE
Add maintenance document

### PR DESCRIPTION
**Pull request by Byron**
Follow-up of #354 .

This document describes common workflows which should help current and new maintainers alike.

Please note that the initial version was created by deducing the release process, and reviewers
are invited to make the necessary corrections or add more workflows as they see fit.

### CODE REVIEW

**Comment by Byron** [comment]
Merging this as it doesn't seem too wrong, while it can be updated as necessary.

File: MAINTENANCE.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+### Creating a new release
+
+#### Artifacts
+
+* a tag of the version number
+* a new [crate version](https://crates.io/crates/flate2/versions)
+
+#### Process
+
+To generated all the artifacts above, one proceeds as follows:
```
Comment (JohnTitor) [inline MAINTENANCE.md:12]:
```suggestion
To generate all the artifacts above, one proceeds as follows:
```

**Review by JohnTitor — APPROVED** [review]
Sorry for the delay, LGTM with one nit!

**Comment by Byron** [comment]
Thanks a lot :)!

**Pull request merged**